### PR TITLE
Upgrade to Halogen 0.5.0; fix #24

### LINKED ISSUE #24: Upgrade to new Halogen

### PR DESCRIPTION
**Pull request by jonsterling**
- upgrade dependencies and build infrastructure
- support new component model
- fix warnings, minor cleanup

~~:zap: This is ready for review, but not merge; once `purescript-halogen v0.5.0` is released, I will update this PR accordingly. :zap:~~

### CODE REVIEW

File: docs/Data/Json/JTable.md [hunk 1]
```diff
@@ -3,19 +3,19 @@
 #### `noStyle`
 
 ``` purescript
-noStyle :: TableStyle
+noStyle :: forall p. TableStyle p
```
Comment (garyb) [inline docs/Data/Json/JTable.md]: Is this due to my recent changes? :/

Comment (jonsterling) [inline docs/Data/Json/JTable.md]: Nah, it's unrelated :) I just needed this to be open-ended within the module it was defined.

**Comment by jonsterling** [comment]
I've updated this PR to adapt to last night's Halogen changes, and also simplified the types that I had made more complicated before. Likewise, this will be ready to merge after this PR is updated to reflect the (impending) Halogen 0.5.0 release.

**Comment by jonsterling** [comment]
OK, I think that this is ready for final review & merge, now that I have updated it to point at the released version of Halogen.

File: src/Data/Json/JTable/Internal.purs [hunk 2]
```diff
@@ -84,48 +93,54 @@ toPrim = foldJsonPrim Just (const Nothing) (const Nothing)
 enumerate :: forall a. List a -> List (Tuple a Int)
 enumerate xs = zipWith Tuple xs (range zero $ (length xs - one))
 
-renderJTableRaw :: JTableOpts -> Json -> Markup
+renderJTableRaw :: forall a. JTableOpts -> Json -> Markup
```
Comment (cryogenian) [inline src/Data/Json/JTable/Internal.purs]: `forall a.`

Comment (jonsterling) [inline src/Data/Json/JTable/Internal.purs]: thank you for catching it!

**Comment by cryogenian** [comment]
:+1:

**Comment by jonsterling** [comment]
Just realized I need to update the README.

**Pull request merged**